code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -2484,3 +2494,57 @@
 	damage *= bonus
 	hit_and_run = 1 //reset the hit and run bonus
 	return damage
+
+/mob/living/carbon/Xenomorph/Hivelord/proc/hivelord_sting(mob/living/H)
+
+	if(!check_state())
+		return
+
+	if(world.time < last_hivelord_sting + HIVELORD_STING_COOLDOWN) //Sure, let's use this.
+		to_chat(src, "<span class='xenodanger'>You are not ready to Sting again. It will be ready in [(last_hivelord_sting + HIVELORD_STING_COOLDOWN - world.time) * 0.1] seconds.</span>")
+		return
+
+	if(stagger)
+		to_chat(src, "<span class='xenowarning'>You try to sting but are too disoriented!</span>")
+		return
+
+	if(!H.can_zombie() )
+		to_chat(src, "<span class='xenowarning'>Your sting won't affect this corpse. A larva took the essence, it's too decayed, or simply too fresh.</span>")
+		return
+
+	if(!Adjacent(H))
+		if(world.time > (recent_notice + notice_delay)) //anti-notice spam
+			to_chat(src, "<span class='xenowarning'>You can't reach this target!</span>")
+			recent_notice = world.time //anti-notice spam
+		return
+
+	if(!check_plasma(150))
+		return
+	last_hivelord_sting = world.time
+	use_plasma(150)
+
+	addtimer(CALLBACK(src, .hivelord_sting_cooldown), HIVELORD_STING_COOLDOWN)
+
+	recurring_injection(H, "necrovirus")
```

Comment:
this is essentially calling
`recurring_injection(mob/living/H, toxin1 = "necrovirus", toxin2 = null, larva = FALSE, virus1 = "necrovirus", count = 2)`